src/main/java/frc/robot/utilities/VisionModule.java

```diff
@@ -0,0 +1,109 @@
+package frc.robot.utilities;
+
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.LinearFilter;
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotContainer;
+
+import javax.annotation.Nullable;
+
+//import static frc.robot.Constants.Vision.*;
+
+public class VisionModule extends SubsystemBase {
+    private static NetworkTable visionTable = NetworkTableInstance.getDefault().getTable("chameleon-vision").getSubTable("ps3");
+    private static NetworkTableEntry visionAngle = visionTable.getEntry("targetYaw");
+    private static NetworkTableEntry visionPose = visionTable.getEntry("targetPose");
+    private static NetworkTableEntry visionValid = visionTable.getEntry("isValid");
+    public static NetworkTableEntry leds = visionTable.getEntry("leds");
+
+    private static LinearFilter filter = LinearFilter.movingAverage(10);
+    private static Double filteredDistance = null;
+
+    /**
+     * @return the angle to the target from the vision network table.
+     */
+    public static double getVisionAngle() {
+        return visionAngle.getDouble(0);
+    }
+
+    /**
+     * @return whether the vision sees the target
+     */
+    public static boolean targetSeen() {
+        return visionValid.getBoolean(false);
+    }
+
+    public static void setLEDs(boolean on) {
+        leds.setBoolean(on);
+    }
+
+    @Nullable
+    public static Pose2d getPose() {
+        Double[] pose = visionPose.getDoubleArray(new Double[]{});
+        if (pose.length == 0) {
+            return null;
+        }
+        return new Pose2d(pose[0], pose[1], Rotation2d.fromDegrees(pose[2]));
+    }
+
+    /**
+     * @return The distance the camera sees, in meters.
+     */
+    @Nullable
+    public static Double getTargetRawDistance() {
+        Pose2d pose = getPose();
+        if (pose == null) return null;
+        return Math.sqrt(Math.pow(pose.getTranslation().getX(), 2) + Math.pow(pose.getTranslation().getY(), 2));
```

Comment:
Use [`Math.hypot`](https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#hypot-double-double-) instead:
```suggestion
        return Math.hypot(pose.getTranslation().getX(), pose.getTranslation().getY());
```